tests/test_sequence.py

```diff
@@ -137,6 +137,280 @@ def test_magnetic_field():
+                    clock_period=4,
+                    phase_jump_time=500,
+                    max_duration=2**26,
+                ),
+            ),
+        ),
+    )
+
+    reg = Register.square(3, 5, prefix="q")
+
+    # Device checkout
+    seq = Sequence(reg, Chadoq2)
+    with pytest.warns(
+        UserWarning,
+        match="Switching a sequence to the same device"
+        + " returns the sequence unchanged.",
+    ):
+        seq.switch_device(Chadoq2)
+    assert seq.switch_device(Chadoq2)._device == Chadoq2
+
+    seq = Sequence(reg, Chadoq2)
+    with pytest.raises(
+        NotImplementedError,
+        match="Switching the device of a sequence to one"
+        + " with reusable channels is not supported.",
+    ):
+        seq.switch_device(MockDevice)
+
+    seq = Sequence(reg, MockDevice)
+    seq.declare_channel("ising", "rydberg_global")
+    with pytest.raises(
+        ValueError,
+        match="Device macth failed because of different rydberg levels.",
+    ):
+        seq.switch_device(IroiseMVP)
+
+    # Different Channels basis
+    test_device1.change_rydberg_level(IroiseMVP.rydberg_level)
+    seq = Sequence(reg, test_device1)
+    seq.declare_channel("digital", "raman_global", "q1")
+    with pytest.raises(
+        TypeError,
+        match="No match for channel digital with the"
+        + " right basis and addressing.",
+    ):
+        seq.switch_device(IroiseMVP)
+
+    # Different addressing channels
+    test_device1.change_rydberg_level(test_device2.rydberg_level)
+    seq = Sequence(reg, test_device1)
+    seq.declare_channel("ising", "raman_global")
+
+    with pytest.raises(
+        TypeError,
+        match="No match for channel ising with the"
+        + " right basis and addressing.",
+    ):
+        seq.switch_device(test_device2)
+
+    # Channel addressing and basis issue
+    test_device2.change_rydberg_level(IroiseMVP.rydberg_level)
+    seq = Sequence(reg, test_device2)
+    seq.declare_channel("ising", "rmn_local")
+
+    with pytest.raises(
+        TypeError,
+        match="No match for channel ising with the"
+        + " right basis and addressing.",
+    ):
+        seq.switch_device(IroiseMVP)
+
+    # Strict: Jump_phase_time & CLock-period criteria
+    # Jump_phase_time check 1: post_phase_shift not nill
+    r_interatomic = Chadoq2.rydberg_blockade_radius(2 * np.pi)
+    reg = Register.square(3, r_interatomic, prefix="q")
+    seq = Sequence(reg, test_device3)
+    rise = Pulse.ConstantDetuning(
+        RampWaveform(250, 0.0, 2.3 * 2 * np.pi), -4 * np.pi, 0.0, 0.5
+    )
+    sweep = Pulse.ConstantAmplitude(
+        2.3 * 2 * np.pi, RampWaveform(400, -4 * np.pi, 4 * np.pi), 1.0, 0.6
+    )
+    fall = Pulse.ConstantDetuning(
+        RampWaveform(500, 2.3 * 2 * np.pi, 0.0), 4 * np.pi, 0.0, 0.7
+    )
+    seq.declare_channel("ising", channel_id="rydberg_global")
+    seq.add(rise, "ising")
+    seq.add(sweep, "ising")
+    seq.add(fall, "ising")
```

Comment:
You should turn this into a fixture, it doesn't make sense to be repeating the same code block